### PR TITLE
Add a Glossary page

### PR DESCRIPTION
**Pull request by cnunciato**
Still a WIP, and super-simple -- we'll want to add some navigation as well (anchoring) and grouping of terms, but this should give us something to start iterating on.

The page isn't linked from anywhere yet, so you'll need to tweak the URL manually to tack on `/learn/glossary` to find it. 

Fixes: https://github.com/pulumi/marketing/issues/194

### CODE REVIEW

File: themes/default/layouts/learn/glossary/list.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            {{ $page := .}}
+
+            {{ range $terms }}
+                {{ $anchor := .term | anchorize }}
+                <div>
+                    <ul>
+                        <li><a href="#{{ $anchor }}">{{ .term }}</a></li>
+                    </ul>
+                </div>
+            {{ end }}
```
Comment (cnunciato) [inline themes/default/layouts/learn/glossary/list.html:27]: Little thing, but this results in a bunch of `ul`s stacked on top of one another, but we probably want one `ul` to wrap all the terms:

```suggestion
            <ul>
                {{ range $terms }}
                    {{ $anchor := .term | anchorize }}
                    <li><a href="#{{ $anchor }}">{{ .term }}</a></li>
                {{ end }}
            </ul>
```

**Review by cnunciato — COMMENTED** [review]
@katcosgrove This is **awesome**. I can't approve 'cause I'm the submitter, so feel free to accept whatever suggestions you want and approve and merge when you're ready. 🎉 

Were you thinking we'd link from the main nav under Learn for or linking somehow from the Learn home page? If a main-nav link is easiest, we can do that for now, but we might want to give this a proper position on the Learn home page as a follow-up.

File: themes/default/layouts/learn/glossary/single.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        <div class="md:w-7/12 my-8">
+            <header class="container mx-auto">
+                {{ partial "docs/breadcrumb.html" . }}
+                <h1>{{ .Title }}</h1>
+            </header>
+            <section class="container mx-auto my-8">
+                {{ .Content }}
+            </section>
+            <p>
+                <a href="{{ relref . "/learn/glossary" }}">&larr; Go back to the Glossary/a>
```
[the remark this answers is not in the document]
Comment (katcosgrove) [reply themes/default/layouts/learn/glossary/single.html:17]: Yes, I'd like to link from the main nav under learn for now, please!

**Review by katcosgrove — APPROVED** [review]
/lgtm

**Pull request merged**